### ISSUE TITLE
SAMRAI linking error on develop branch

### ISSUE THREAD

**Issue by benjaminjeliot** [UK-MAC/CleverLeaf_ref#2]
I encountered an error on the linking step when building the Cleverleaf develop branch.

>In function 'SAMRAI::mesh::GriddingAlgorithm::regridFinerLevel_createAndInstallNewLevel(int, double, boost::shared_ptr<SAMRAI::hier::Connector>&, boost::shared_ptr<SAMRAI::hier::Connector const>, boost::shared_ptr<SAMRAI::hier::BoxLevel>)':
> GriddingAlgorithm.C:(.text+0x905f): undefined reference to 'SAMRAI::xfer::PatchLevelInteriorFillPattern::PatchLevelInteriorFillPattern()'

It would successfully link if I reordered the SAMRAI libraries in src/CMakeLists.txt so that SAMRAI_XFER_LIBRARIES follows SAMRAI_MESH_LIBRARIES, i.e.

```
  ${SAMRAI_ALGS_LIBRARIES}
  ${SAMRAI_APPU_LIBRARIES}
  ${SAMRAI_MATH_LIBRARIES}
  ${SAMRAI_MESH_LIBRARIES}
  ${SAMRAI_XFER_LIBRARIES}
  ${SAMRAI_GEOM_LIBRARIES}
  ${SAMRAI_PDAT_LIBRARIES}
  ${SAMRAI_SOLV_LIBRARIES}
  ${SAMRAI_HIER_LIBRARIES}
  ${SAMRAI_TBOX_LIBRARIES}
```

**Comment by davidbeckingsale** [comment]
Hey @benjaminjeliot, I've run into that before too - I will get this fixed.

**Comment by davidbeckingsale** [comment]
Closed by #3
